integration/app_integration_test.go

```diff
@@ -1092,6 +1111,29 @@ func (p *pack) createAppSession(t *testing.T, publicAddr, clusterName string) st
 	err = json.NewDecoder(resp.Body).Decode(&casResp)
 	require.NoError(t, err)
 
+	require.Eventually(t, func() bool {
+		events, _, err := p.rootCluster.Process.GetAuthServer().SearchEvents(
+			time.Now().Add(-time.Hour),
+			time.Now().Add(time.Hour),
+			apidefaults.Namespace,
+			[]string{events.AppSessionStartEvent},
+			1,
+			types.EventOrderDescending,
+			"",
+		)
+		require.NoError(t, err)
+		if len(events) == 0 {
+			return false
+		}
+
+		startSessionEvent, ok := events[0].(*apievents.AppSessionStart)
+		require.True(t, ok)
+		require.Equal(t, uri, startSessionEvent.AppURI)
+		require.Equal(t, publicAddr, startSessionEvent.AppPublicAddr)
+		require.Equal(t, name, startSessionEvent.AppName)
```

Comment:
Let's compare entire event objects instead of individual fields. Would make the test more robust.